src/S3/S3Client.php

```diff
@@ -678,9 +678,11 @@ private function processEndpointV2Model()
         foreach($definition['operations'] as &$operation) {
             if (isset($operation['http']['requestUri'])) {
                 $requestUri = $operation['http']['requestUri'];
-                $requestUri = str_replace('/{Bucket}', '/', $requestUri);
-                $requestUri = str_replace('//', '/', $requestUri);
-                $requestUri = str_replace('/?', '?', $requestUri);
+                if ($requestUri === "/{Bucket}") {
+                    $requestUri = str_replace('/{Bucket}', '/', $requestUri);
+                } else {
+                    $requestUri = str_replace('/{Bucket}', '', $requestUri);
```

Comment:
just trying to undrstand this- why do we only have a trailing slash when the uri is /{Bucket}?

Reply:
We/S3 expect there to be a trailing slash on operations (CreateBucket, DeleteBucket, etc.) where there is no additional path or query.  For the others, if there's more path string beyond `{Bucket}`, a forward slash will already be present, or the query will just be appended.